src/api/sms-api/sms.ts

```diff
@@ -0,0 +1,111 @@
+import { z } from 'zod';
+import { http, HttpResponse } from 'msw';
+import { mockKandidatlisteMedStilling } from '../../../mock/kandidat-api/mockKandidatliste';
+import { mockVeileder } from '../../../mock/mockVeileder';
+import { fetchJson, postJson } from '../../kandidat/api/fetchUtils';
+
+const varselStillingEndepunkt = (stillingId: string) =>
+    `/kandidatvarsel-api/varsel/stilling/${stillingId}`;
+const varselQueryEndepunkt = '/kandidatvarsel-api/varsel/query';
+
+export enum Meldingsmal {
+    VurdertSomAktuell = 'VURDERT_SOM_AKTUELL',
+    FunnetPassendeStilling = 'PASSENDE_STILLING',
+    Jobbarrangement = 'PASSENDE_JOBBARRANGEMENT',
+    // EtterspurtPgaKorona = 'etterspurt_pga_korona',
+    // Webinar = 'webinar',
+}
+
+export enum ServerSmsStatus {
+    UnderUtsending = 'UNDER_UTSENDING',
+    Sendt = 'SENDT',
+    Feil = 'FEIL',
+}
+
+const SmsStatusSchema = z
+    .literal(ServerSmsStatus.UnderUtsending)
+    .or(z.literal(ServerSmsStatus.Sendt))
+    .or(z.literal(ServerSmsStatus.Feil));
+
+const SmsSchema = z.object({
+    id: z.string(),
+    opprettet: z.string(),
+    fnr: z.string(),
+    stillingId: z.string(),
+    navIdent: z.string(),
+    status: SmsStatusSchema,
+});
+
+export type Sms = z.infer<typeof SmsSchema>;
+
+const SmsArraySchema = z.array(SmsSchema);
+
+/* Ikke skrevet om til useSwr siden disse brukes i redux-reducer */
+
+type smserForStillingRequest = { stillingId: string };
+export const fetchSmserForStilling = async ({
+    stillingId,
+}: smserForStillingRequest): Promise<Sms[]> =>
+    SmsArraySchema.parse(await fetchJson(varselStillingEndepunkt(stillingId)));
+
+type smserForKandidatRequest = { fnr: string };
+export const fetchSmserForKandidat = async ({ fnr }: smserForKandidatRequest): Promise<Sms[]> =>
```

Comment:
Vi har brukt useswr i det siste, burde det også brukes her på samme format som for de andre kallene?

Reply:
Skrevet om til useswr :)